### PR TITLE
Setup black, isort and pytest

### LINKED ISSUE #7: Hook up travis
https://github.com/simon-weber/gpsoauth/pull/6 added tests that could be run via travis.

### PR DESCRIPTION
**Pull request by KapJI**
- Run them using `pre-commit`.
- Setup CI job on Github Actions which runs all `pre-commit` checks.
- Remove `tox` config. Poetry already provides venv and different python versions are more easily tested on Github Actions.

Another part of #26.

Closes #7

### CODE REVIEW

**Comment by KapJI** [comment]
I checked multiple times that path to CI config is correct, I think it will run only after merging this.

File: pyproject.toml [hunk 1]
```diff
@@ -21,13 +21,22 @@ homepage = "https://github.com/simon-weber/gpsoauth"
...
+black = "^20.8b1"
+isort = "^5.8.0"
+pytest = "^6.2.2"
 
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.isort]
+profile = "black"
```
Comment (simon-weber) [inline pyproject.toml:40]: ah, cool; I'm guessing this avoids the conflicts I've seen occasionally.

Comment (KapJI) [reply pyproject.toml:40]: Yes, some tools need to be configured to be compatible with `black` 🙂
https://black.readthedocs.io/en/stable/compatible_configs.html

**Comment by simon-weber** [comment]
looks good! I'll set up Actions and try to get it running on a copy of this branch.

**Comment by KapJI** [comment]
Not sure, but probably it will only work after being committed to the default branch (`master`).

**Comment by simon-weber** [comment]
Looks good. Here's a test run with no changes: https://github.com/simon-weber/gpsoauth/runs/2247269621

And here's one with some failures from --no-verify: https://github.com/simon-weber/gpsoauth/runs/2247329653

**Comment by simon-weber** [comment]
And here's the first run on master! https://github.com/simon-weber/gpsoauth/actions/runs/709002120 🎉

**Pull request merged**